src/commands/deploy/metadata/quick.ts

```diff
@@ -0,0 +1,129 @@
+const messages = Messages.loadMessages('@salesforce/plugin-deploy-retrieve', 'deploy.metadata.quick');
+
+export type DeployMetadataQuickResult = {
+  files?: FileResponse[];
+  jobId: string;
+  tests?: TestResults;
+};
+
+export default class DeployMetadataQuick extends SfCommand<DeployMetadataQuickResult> {
+  public static readonly description = messages.getMessage('description');
+  public static readonly summary = messages.getMessage('summary');
+  public static readonly examples = messages.getMessages('examples');
+  public static readonly requiresProject = true;
+  public static flags = {
+    'job-id': Flags.salesforceId({
+      char: 'i',
+      startsWith: '0Af',
+      description: messages.getMessage('flags.job-id.description'),
+      summary: messages.getMessage('flags.job-id.summary'),
+      exactlyOne: ['use-most-recent', 'job-id'],
+    }),
+    'use-most-recent': Flags.boolean({
+      char: 'r',
+      description: messages.getMessage('flags.use-most-recent.description'),
+      summary: messages.getMessage('flags.use-most-recent.summary'),
+      exactlyOne: ['use-most-recent', 'job-id'],
+    }),
+    wait: Flags.duration({
+      char: 'w',
+      summary: messages.getMessage('flags.wait.summary'),
+      description: messages.getMessage('flags.wait.description'),
+      unit: 'minutes',
+      defaultValue: 33,
+      helpValue: '<minutes>',
+      min: 1,
+    }),
+  };
+
+  public static envVariablesSection = toHelpSection('ENVIRONMENT VARIABLES', EnvironmentVariable.SF_TARGET_ORG);
+
+  private org: Org;
+
+  public async run(): Promise<DeployMetadataQuickResult> {
+    const flags = (await this.parse(DeployMetadataQuick)).flags;
+    const cache = await DeployCache.create();
+    let jobId = flags['job-id'] || '';
+
+    if (flags['use-most-recent']) {
+      jobId = cache.getLatestKey();
+      if (!jobId) throw messages.createError('error.NoRecentJobId');
+    }
+
+    if (!cache.has(jobId)) {
+      throw messages.createError('error.InvalidJobId', [jobId]);
+    }
+
+    const deployOpts = cache.get(jobId);
+    this.org = await Org.create({ aliasOrUsername: deployOpts['target-org'] });
+
+    const deploy = await this.org
+      .getConnection()
+      .deployRecentValidation({ id: jobId, rest: deployOpts.api === API.REST });
+    const componentSet = await buildComponentSet({ ...deployOpts, wait: flags.wait });
+
+    this.log(getVersionMessage('Deploying', componentSet, deployOpts.api));
+    this.log(`Deploy ID: ${jobId}`);
+
+    const result = await this.poll(jobId, flags.wait, componentSet);
+
+    if (!this.jsonEnabled()) {
+      displayDeployResults(result, deployOpts['test-level'], deployOpts.verbose);
+    }
+
+    await DeployCache.unset(jobId);
+
+    return {
+      jobId,
+      tests: getTestResults(deploy as unknown as MetadataApiDeployStatus),
+      files: result.getFileResponses() || [],
+    };
+  }
+
+  protected catch(error: SfCommand.Error): Promise<SfCommand.Error> {
```

Comment:
This overrides the `catch` method in `SfCommand` which is where we do all the custom formatting for errors. In this case we just want to massage the error message slightly before letting that formatting happen

So generally speaking it should only be overridden when you have to do something to the error before it gets logged. It should also always return `super.catch(err)` so that the error does ultimately get logged to the user

Reply:
Sounds like a great thing to put as a comment in the code.

Reply:
Thats *also* sounds like a useful custom eslint rule to keep people from hurting themselves